src/Build/Evaluation/Expander.cs

```diff
@@ -123,6 +123,141 @@ internal class Expander<P, I>
         where P : class, IProperty
         where I : class, IItem
     {
+        /// <summary>
+        /// A helper struct wrapping a <see cref="SpanBasedStringBuilder"/> and providing file path conversion
```

Comment:
Are there any other places that might benefit from this?

Reply:
It may find its use in other parts of `Expander`, although a cursory look through metadata and item expansion does not really suggest that they are suffering from the same problems. They actually look reasonable.

If I find other places where this may be useful I'll pull it out into its own top-level type, add functionality as needed, maybe rename, etc.